submissions/SamVal007/Tiny-JS-OOP/index.js

```diff
@@ -0,0 +1,63 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: https://samval007.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Person {
+  constructor(species, gender, name, speach, legs) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.speach = speach;
+    this.legs = legs;
+  }
+
+  saySpeach() {
+    const sayingSpeach = [
+      this.speach,
+      `I am a ${this.species}.`,
+      `My name is ${this.name}.`,
+      `My gender is ${this.gender}.`,
+      `I have ${this.legs} legs.`
+    ];
+    return sayingSpeach.join(' ');
+  }
+
+}
+
+class Dog extends Person {
+  constructor(name, gender, speach, legs = 4) {
+    super("dog", name, gender, speach, legs);
+  }
+}
+class Cat extends Person {
+  constructor(name, gender, speach, legs = 4) {
+    super("cat", name, gender, speach, legs);
+  }
+}
+class Human extends Person {
+  constructor(name, gender, speach, friends, legs = 2, hands = 2) {
+    super("human", name, gender, speach, friends, legs, hands);
```

Comment:
What `super` call actually does?
What wrong here and how could you fix this?

Reply:
As I understand, when we create a child class by extending parent class, we call super() in order to initialize general fields and avoid code duplication. Custom fields from child we need initialize manually using keyword "this".